### ISSUE TITLE
How to get ATT-mono in Table 3

### ISSUE THREAD

**Issue by yhcc** [muhaochen/BilDRL#3]
I tried to use embeddings in bilbowa.zip file to retrieve the corresponding word (cosine distance and Ecludean distance have been tried), however, the peformance is far behind yours. Although for monolingual setting (for example, English description to English word), I achieve 0.8 top1 accuracy, I used the top1 English word to retrieve the Spanish word based on the word embeddings, the top1 accuracy is only 0.2. Therefore, would you please give more details on how you implement this Att-mono in Table 3? Thanks in advance.

**Comment by muhaochen** [comment]
That is exactly what we did for the monolingual baseline method:
1. Train the mapping from the English description to the English lexeme.
2. Use the bilingual word embedding to get the foreign lexemes.
As being discussed in the paper, you would expect this method to perform not that well.

**Comment by yhcc** [comment]
Thanks for your timely reply. Since you provided a translations of the target word in the first column, therefore, I used this translation to retrieve the corresponding target word through the provided embeddings. However, the following are my results:
tgt_src: top1 top10
en_es : 0.224 0.395
en_fr : 0.335 0.491
fr_en : 0.162 0.279
es_en : 0.144 0.236
While, ATT-mon has the following results:
tgt_src: top1 top10
en_es : 0.346 0.422
en_fr :  0.228 0.336
fr_en : 0.274 0.390
es_en : 0.394 0.486
As you can see, your performance is  better than using translation to retrieve words. Since we also have to implement a baseline like you presented in your paper, we don't want to fall too much behind your baseline. Can you provide some help? Thanks~

**Comment by muhaochen** [comment]
Ok, let me give my first guess. There was a constraint to the search space as described in the Evaluation Protocals:  
We limit the vocabularies to all words that appear in the Wikt3l dataset, which involve around 45k English words, 44k French words and 36k Spanish words.
Did you search among the appearing words or the entire vocabulary (500k words each language) of the pre-trained bilingual word embeddings?

**Comment by yhcc** [comment]
Sounds like reasonable. I guess that will be the answer. Thanks.

**Comment by muhaochen** [comment]
Ok, just to make sure by the way, all words that appear in the Wikt3l dataset *includes* also words in the definitions, not just targets.

**Comment by yhcc** [comment]
When retrieve the results, did you use the word rank list predicted by the monlingual model to retrieve the nearest words, or did you use the top predict word to find words and ordered by the distance between this word?
